akka-docs/rst/scala/persistence-schema-evolution.rst

```diff
@@ -0,0 +1,386 @@
+being delivered to a recovering ``PersistentActor`` is pretty simple, as one can simply filter them out in an ``EventAdapter``:
+
+
+.. figure:: ../images/persistence-drop-event.png
+   :align: center
+
+   The ``EventAdapter`` can drop old events (**O**) by emitting an empty :class:`EventSeq`.
+   Other events can simply be passed through (**E**).
+
+This however does not address the underlying cost of having to deserialize all the events during recovery,
+even those which will be filtered out by the adapter. In the next section we will improve the above explained mechanism
+to avoid deserializing events which would be filtered out by the adapter anyway, thus allowing to save precious time
+during a recovery containing lots of such events (without actually having to delete them).
+
+**Improved solution - deserialize into tombstone:**
+
+In the just described technique we have saved the PersistentActor from receiving un-wanted events by filtering them
+out in the ``EventAdapter``, however the event itself still was deserialized and loaded into memory.
+This has two notable downsides:
+
+  - first, that the deserialization was actually performed, so we spent some of out time budget on the
+    deserialization, event though the event does not contribute anything to the persistent actors state.
+  - second, that we are *unable to remove the event class* from the system – since the serializer still needs to create
+    the actuall instance of it, as it does not know it will not be used.
+
+The solution to these problems is to use a serializer that is aware of that event being no longer needed, and can notice
+this before starting to deserialize the object. This can for example be implemented by using an ``SerializerWithStringManifest``
+(documented in depth in :ref:`string-manifest-serializer-scala`). By looking at the string manifest, the serializer can notice
+that the type is no longer needed, and skip the deserialization all-together:
+
+.. figure:: ../images/persistence-drop-event-serializer.png
+   :align: center
+
+   The serializer is aware of the old event types that need to be skipped (**O**), and can skip deserializing them alltogether
+   by simply returning a "tombstone" (**T**), which the EventAdapter converts into an empty EventSeq.
+   Other events (**E**) can simply be passed through.
+
+The serializer detects that the string manifest points to a removed event type and skips attempting to deserialize it:
+
+.. includecode:: code/docs/persistence/PersistenceSchemaEvolutionDocSpec.scala#string-serializer-skip-deleved-event-by-manifest
+
+The EventAdapter we implemented is aware of ``EventDeserializationSkipped`` events (our "Tombstones"),
+and emits and empty ``EventSeq`` whenever such object is encoutered:
+
+.. includecode:: code/docs/persistence/PersistenceSchemaEvolutionDocSpec.scala#string-serializer-skip-deleved-event-by-manifest-adapter
+
+.. _detach-domain-from-data-model-scala:
+
```

Comment:
somewhere in this section it should also be described that it is possible to completely remove the class file, and that the serialization config for that class can be amended to the serializer is only registered for fromBinary (requiring a WithStringManifestSerializer)